shenyu-sdk/shenyu-sdk-core/src/main/java/org/apache/shenyu/sdk/core/http/AbstractShenyuHttpClient.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.sdk.core.http;
+
+import org.apache.shenyu.common.config.ShenyuConfig;
+import org.apache.shenyu.common.utils.Singleton;
+import org.apache.shenyu.register.common.dto.InstanceRegisterDTO;
+import org.apache.shenyu.register.instance.api.ShenyuInstanceRegisterRepository;
+import org.apache.shenyu.register.instance.core.ShenyuInstanceRegisterRepositoryFactory;
+import org.apache.shenyu.sdk.core.ShenyuRequest;
+import org.apache.shenyu.sdk.core.ShenyuResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public abstract class AbstractShenyuHttpClient implements ShenyuHttpClient {
+
+    private static final Logger log = LoggerFactory.getLogger(AbstractShenyuHttpClient.class);
+
+    private static final String URL_REWRITE_REGEX = "http:\\/\\/[a-z\\d\\.|:]+\\/";
+
+    private final Retryer retryer;
+
+    private final ShenyuInstanceRegisterRepository registerRepository;
+
+    private final ShenyuConfig.RegisterConfig sdkConfig;
+
+    public AbstractShenyuHttpClient() {
+        this.sdkConfig = Optional.ofNullable(Singleton.INST.get(ShenyuConfig.class)).orElse(new ShenyuConfig()).getSdk();
```

Comment:
this parameter always nonnull,you should just shenyuConfig.sdk

Reply:
done